### ISSUE TITLE
Change API to accept seconds instead of milliseconds

### ISSUE THREAD

**Issue by jaxley** [helmetjs/hsts#7]
You tell the reader to refer to the HSTS spec in your README, but the API provided doesn't follow the specification in that it requires you to pass in the max-age in milliseconds (even this is not very clear about the API, except for a single example in the README).  This could lead to errors of a thousand fold in the expiration for people familiar with HSTS but not with your API deviations from the spec.

From the spec:

> "The REQUIRED "max-age" directive specifies the number of seconds,
>    after the reception of the STS header field, during which the UA
>    regards the host (from whom the message was received) as a Known HSTS
>    Host."

**Comment by EvanHahn** [comment]
This one is tricky because JavaScript usually deals with millisecond intervals.

If the max-age is in seconds, it's consistent with the spec but inconsistent with JavaScript. If the max-age is in milliseconds, it's consistent with JavaScript but inconsistent with the spec.

IMO, the solution to this is to make the documentation better (probably by adding a note that says "this is different from the spec!"). What do you think?

**Comment by jaxley** [comment]
I'm not sure how JavaScript timeframes are even relevant?  The value
provided is passed directly to the http header and ultimately has to be in
seconds.  It's about what the used agents need.

Or is there another angle I'm missing where having it in milliseconds makes
it easier to consume in JavaScript apps?

I'm not a fan of using documentation to address a potentially dangerous
situation such as this where the likely error will result in policy being 3
orders of magnitude off expectations.

Jason

**Comment by EvanHahn** [comment]
I think of `max-age` as a duration. Durations in JavaScript are in milliseconds (I'd cite [ms](https://github.com/rauchg/ms.js), [Moment's `diff`](http://momentjs.com/docs/#/displaying/difference/), and what happens when you subtract one `Date` from another). Perhaps that's not the right way of looking at it—perhaps `max-age` is something else?

I think the `max-age` can be off by 3 orders of magnitude in either case. As things are today, this middleware can set the header for 0.1% of what a developer expected. Your suggestion can set the header to 1000% of what was expected.

I certainly think the documentation needs to be improved to highlight this change, but I'm not (yet?) convinced that changing the argument is any less dangerous.

Thoughts?

**Comment by EvanHahn** [comment]
Whoops, didn't mean to close.

**Comment by jaxley** [comment]
I still don't see why it needs to have anything to do with what might be "normal" in JavaScript (and it seems normal is just that ECMAScript represents dates as milliseconds - but milliseconds since the UNIX Epoch). The specification is not written for javascript and is in seconds.  Someone who is intending to set a max-age for HSTS is going to want to set a max-age _as understood within the domain of HSTS_, which is in seconds, not milliseconds.  

HSTS expirations won't have millisecond accuracy and to expose an API as if they do is also misleading.  

Not sure what you mean by a duration in JavaScript being seconds - you simply mean doing a difference between two Date Objects?  That is a delta in milliseconds.  And one of the modules you referenced is specifically to perform millisecond conversions, so it would obviously be biased toward milliseconds.  But also not unexpected given that ECMAScript uses milliseconds internally.  The other one has APIs to get the moments in various other measurements, not just milliseconds.  

Couldn't one make an equivalent case that using UNIX Epoch references to denote the max-age (e.g. d.setDate(d.getDate() + 365)), which returns UNIX Epoch time) would be just as valid?  It's even the internal representation of a date. 

I could only see a possible argument for doing something unexpected if:

1. it greatly simplified the code using the API or consuming it (unlikely as most of these would be simple absolute hard-coded values and not dynamically constructed, additionally we're talking about the difference between a possible divide-by-1000 so doesn't save much on efficiency)
1. it would not cause confusion by all the other APIs or alternatives in the web world (such as directly emitting the values in headers yourself) that set this.  e.g. Spring security:  http://docs.spring.io/spring-security/site/docs/current/reference/html/headers.html#headers-hsts

My $0.02.

**Comment by EvanHahn** [comment]
I brought this up with the frontend team at Braintree and took some votes.

- `max-age` as milliseconds: 3 votes (with one more "yes, but only if...")
- `max-age` as seconds: 7 votes (with one "maybe" hand gesture)

Based on that alone, I think we should switch to seconds. There were a few other ideas, too.

This kind of API was also proposed:

```js
app.use(hsts({
  maxAgeInSeconds: 1000
})

app.use(hsts({
  maxAgeInMilliseconds: 1000000
})
```

What do you think of that?

**Comment by frodsan** [comment]
👍 to change to seconds. Coming from other language, I expected to set maxAge as seconds (as the spec), but I ended up with:

```
Strict-Transport-Security: max-age=15552; includeSubDomains
```

**Comment by EvanHahn** [comment]
Okay—I'm convinced that this should be done. I'm going to rename this issue to "Change API to accept seconds instead of milliseconds".

Would anyone here like to submit a pull request for review? No need to do a major version bump—I'll take care of that.

If nobody takes it soon, I'll try to get to it.

**Comment by EvanHahn** [comment]
See #8.

**Comment by EvanHahn** [comment]
This has been merged into the `2.0` branch and will be released with the next major version of this module.

Thanks for your feedback everyone!
